hawkular-dmr-client/src/main/java/org/hawkular/dmrclient/DeploymentJBossASClient.java

```diff
@@ -38,22 +38,27 @@ public DeploymentJBossASClient(ModelControllerClient client) {
         super(client);
     }
 
-    public void enableDeployment(String name) throws Exception {
-        enableDisableDeployment(name, true);
+    public void enableDeployment(String name, Set<String> serverGroups) throws Exception {
+        enableDisableDeployment(name, true, serverGroups);
     }
 
-    public void disableDeployment(String name) throws Exception {
-        enableDisableDeployment(name, false);
+    public void disableDeployment(String name, Set<String> serverGroups) throws Exception {
+        enableDisableDeployment(name, false, serverGroups);
     }
 
-    private void enableDisableDeployment(String name, boolean enable) throws Exception {
+    private void enableDisableDeployment(String name, boolean enable, Set<String> serverGroups) throws Exception {
         DeploymentManager dm = DeploymentManager.Factory.create(getModelControllerClient());
         DeploymentResult result;
 
         if (enable) {
-            result = dm.deployToRuntime(SimpleDeploymentDescription.of(name));
+            SimpleDeploymentDescription sdd = SimpleDeploymentDescription.of(name);
+            sdd.addServerGroups(serverGroups);
+            result = dm.deployToRuntime(sdd);
         } else {
-            result = dm.undeploy(UndeployDescription.of(name).setRemoveContent(false));
+            UndeployDescription ud = UndeployDescription.of(name);
+            ud.addServerGroups(serverGroups);
```

Comment:
what if serverGroups is null? Does the API accept that? Or might be best to do:

sdd.addServerGroups(serverGroups != null ? serverGroups : Collections.emptySet());